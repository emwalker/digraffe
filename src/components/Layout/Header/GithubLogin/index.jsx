// @flow
import React from 'react'

const GithubLogin = () => (
  <summary className="p-2 d-inline">
    <a href="/auth/github" className="mt-1">Log in with GitHub</a>
  </summary>
)

export default GithubLogin
