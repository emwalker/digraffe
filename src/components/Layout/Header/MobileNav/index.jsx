// @flow
import React, { Component } from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { Link } from 'found'
import classNames from 'classnames'

import DigraphLogo from 'components/ui/icons/DigraphLogo'
import Menu from './Menu'
import styles from './styles.module.css'
import type { MobileNav_viewer as Viewer } from './__generated__/MobileNav_viewer.graphql'

type Props = {
  viewer: Viewer,
}

type State = {
  isOpen: boolean,
}

class MobileNav extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      isOpen: false,
    }
  }

  onClick = () => {
    this.setState((prevState) => ({ isOpen: !prevState.isOpen }))
  }

  render = () => (
    <div className={classNames(styles.mobileMenu, 'mobile-menu mb-4')}>
      <div className="mobile-menu-header d-flex px-3 py-2">
        <Link
          to="/"
          className={classNames(
            styles.link,
            'h3 text-normal menu-logo text-gray-dark n-link no-underline d-flex flex-items-center',
          )}
        >
          <div className="mr-2 d-inline-block">
            <DigraphLogo width="32px" height="32px" />
          </div>
          {' '}
          Digraph
        </Link>

        <button
          className="menu-btn btn btn-outline py-1"
          onClick={this.onClick}
          type="button"
        >
          Menu
        </button>
      </div>
      { this.state.isOpen && <Menu viewer={this.props.viewer} /> }
    </div>
  )
}

export const UnwrappedMobileNav = MobileNav

export default createFragmentContainer(MobileNav, {
  viewer: graphql`
    fragment MobileNav_viewer on User {
      isGuest
      ...Menu_viewer
    }
  `,
})
