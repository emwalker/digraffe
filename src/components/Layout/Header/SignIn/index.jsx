// @flow
import React from 'react'
import { Link } from 'found'

const SignIn = () => (
  <Link to="/login" className="px-3 text-gray-dark">Sign in</Link>
)

export default SignIn
