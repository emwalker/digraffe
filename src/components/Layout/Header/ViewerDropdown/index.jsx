// @flow
import React from 'react'

import type { UserType } from 'components/types'

type Props = {
  viewer: UserType,
}

const ViewerDropdown = ({ viewer: { name, avatarUrl } }: Props) => (
  <summary className="p-2 d-inline">
    <a className="mr-3" href="/logout/github">Sign out</a>
    <img
      alt={name}
      className="avatar"
      height="20"
      src={`${avatarUrl}&s=40`}
      width="20"
    />
  </summary>
)

export default ViewerDropdown
