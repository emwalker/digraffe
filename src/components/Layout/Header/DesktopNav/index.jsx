// @flow
import React, { Component } from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { Link } from 'found'

import type { Location, Router } from 'components/types'
import { toEverything } from 'components/navigation'
import DigraphLogo from 'components/ui/icons/DigraphLogo'
import SearchBox from 'components/ui/SearchBox'
import UserNav from './UserNav'
import SignIn from './SignIn'
import { header, primaryLogo, userNav, navLeft, searchBox } from './styles.module.css'
import type { DesktopNav_viewer as Viewer } from './__generated__/DesktopNav_viewer.graphql'

type Props = {
  className?: ?string,
  location: Location,
  router: Router,
  view: any,
  viewer: Viewer,
}

class DesktopNav extends Component<Props> {
  static defaultProps = {
    className: '',
  }

  get className(): string {
    return `Header ${this.props.className || ''}`
  }

  get isGuest(): boolean {
    const { viewer } = this.props

    return viewer ? viewer.isGuest : true
  }

  renderGuestUserNav = () => (
    <>
      <SignIn />
    </>
  )

  renderUserNav = (viewer: Viewer) => <UserNav viewer={viewer} />

  render = () => {
    const { viewer, view, router, location } = this.props

    return (
      <header className={`${header} px-sm-3 px-md-6 px-lg-4`}>
        <nav className={`${navLeft} d-inline-block`}>
          <h1 className="h3 text-normal">
            <a
              href="/"
              className={`${primaryLogo} text-gray-dark n-link no-underline d-flex`}
            >
              <div className="mr-2 d-inline-block">
                <DigraphLogo height="28px" width="28px" fill="#000" />
              </div>

              Digraph
            </a>
          </h1>
        </nav>
        <div className={searchBox}>
          <SearchBox router={router} location={location} view={view} />
        </div>
        <nav className={userNav}>
          <a
            className="text-gray-dark px-2"
            href="https://blog.digraph.app"
          >
            Blog
          </a>
          <Link
            className="text-gray-dark px-2"
            id="recent-activity"
            to="/recent"
          >
            Recent
          </Link>
          <Link
            className="text-gray-dark px-2"
            to={toEverything}
          >
            Everything
          </Link>
          {this.isGuest
            ? this.renderGuestUserNav()
            : this.renderUserNav(viewer)}
        </nav>
      </header>
    )
  }
}

export const UnwrappedDesktopNav = DesktopNav

export default createFragmentContainer(DesktopNav, {
  viewer: graphql`
    fragment DesktopNav_viewer on User {
      isGuest
      ...UserNav_viewer
    }
  `,
})
