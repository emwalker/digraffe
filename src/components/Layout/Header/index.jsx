// @flow
import React from 'react'
import MediaQuery from 'react-responsive'
import classNames from 'classnames'

import DesktopNav from './DesktopNav'
import MobileNav from './MobileNav'
import { header } from './styles.module.css'

type Props = {
  viewer: Object,
  defaultOrganization: Object,
}

export default ({ viewer, defaultOrganization }: Props) => {
  const classes = classNames(header, 'clearfix mb-3 d-flex px-md-6 px-lg-4')

  return (
    <div className={classes}>
      <div style={{ width: '100%' }}>
        <MediaQuery maxWidth={544}>
          <MobileNav viewer={viewer} />
        </MediaQuery>
        <MediaQuery minWidth={544}>
          <DesktopNav
            viewer={viewer}
            defaultOrganization={defaultOrganization}
          />
        </MediaQuery>
      </div>
    </div>
  )
}
