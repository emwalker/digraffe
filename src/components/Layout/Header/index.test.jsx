import React from 'react'
import {shallow} from 'enzyme'

import Header from './index'

const props = {
  viewer: {
    name: 'Frotz',
  },
}

describe('<Header />', () => {
  const wrapper = shallow(<Header {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
