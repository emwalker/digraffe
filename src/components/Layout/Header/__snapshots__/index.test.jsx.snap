// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<header
  className="Header pagehead"
>
  <div
    className="container-lg clearfix"
  >
    <nav
      className="d-lg-flex float-left"
    >
      <h1>
        Digraph
      </h1>
    </nav>
    <div
      className="d-lg-flex float-right mt-1"
    >
      <ul
        className="user-nav d-lg-flex flex-items-center list-style-none"
      >
        <li
          className="dropdown"
        >
          <details
            className="details-overlay details-reset d-none d-lg-flex pl-lg-2 py-2 py-lg-0 flex-items-center"
          >
            <summary
              className="HeaderNavlink name mt-1"
            >
              Frotz
              <span
                className="dropdown-caret"
              />
            </summary>
          </details>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
